Source/FieldSolver/SpectralSolver/SpectralAlgorithms/AvgGalileanAlgorithm.cpp

```diff
@@ -0,0 +1,371 @@
+#include "FieldSolver/SpectralSolver/SpectralAlgorithms/AvgGalileanAlgorithm.H"
+#include "Utils/WarpXConst.H"
+#include <cmath>
+
+using namespace amrex;
+
+/* \brief Initialize coefficients for the update equation */
+AvgGalileanAlgorithm::AvgGalileanAlgorithm(const SpectralKSpace& spectral_kspace,
+                         const DistributionMapping& dm,
+                         const int norder_x, const int norder_y,
+                         const int norder_z, const bool nodal,
+                         const amrex::Array<amrex::Real,3>& v_galilean,
+                         const Real dt)
+     // Initialize members of base classinde
+     : SpectralBaseAlgorithm( spectral_kspace, dm,
+                              norder_x, norder_y, norder_z, nodal )
+{
+    const BoxArray& ba = spectral_kspace.spectralspace_ba;
+
+    // Allocate the arrays of coefficients
+    C_coef = SpectralRealCoefficients(ba, dm, 1, 0);
+    S_ck_coef = SpectralRealCoefficients(ba, dm, 1, 0);
+
+    C1_coef = SpectralRealCoefficients(ba, dm, 1, 0);
+    S1_coef = SpectralRealCoefficients(ba, dm, 1, 0);
+    C3_coef = SpectralRealCoefficients(ba, dm, 1, 0);
+    S3_coef = SpectralRealCoefficients(ba, dm, 1, 0);
+
+    Psi1_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+    Psi2_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+    Psi3_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+
+
+    X1_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+    X2_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+    X3_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+    X4_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+    Theta2_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+
+    A1_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+    A2_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+
+    Rhoold_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+    Rhonew_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+    Jcoef_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+
```

Comment:
In the case of Galilean class, we created a separate function `InitializeSpectralCoefficients`. This was because CUDA compilers usually raise an error when a lambda function is used inside the constructor of the class.

It seems to me that we need to create a similar function here, otherwise the code will not compile on GPU, is that correct?